src/main/java/com/faforever/client/api/TokenRetriever.java

```diff
@@ -59,7 +60,17 @@ private Mono<Void> getExpirationMono(OAuth2AccessToken token) {
   }
 
   public Mono<String> getRefreshedTokenValue() {
-    return refreshedTokenMono.doOnError(this::onTokenError);
+    return refreshedTokenMono.map(OAuth2AccessToken::getTokenValue).doOnError(this::onTokenError);
+  }
+
+  public Mono<String> getRefreshedHmacValue() {
+    return getRefreshedTokenValue().flatMap(tokenValue -> {
+      try {
+        return Mono.just(JWTParser.parse(tokenValue).getJWTClaimsSet().getJSONObjectClaim("ext").get("hmac").toString());
+      } catch (ParseException e) {
+        return Mono.error(e);
+      }
```

Comment:
Why only ParseException? Could in theory also be NullPointerException if the claim is missing